src/Engine/Audio/SteamAudioInterface.hpp

```diff
@@ -15,116 +14,84 @@ class SteamAudioInterface {
 
         /// Destructor.
         ~SteamAudioInterface();
-
-        /// Sets the main context used by Steam Audio.
-        /**
-         * @param logCallback Callback for logging messages. Can be NULL. Default uses stdout.
-         * @param allocCallback Callback for allocating memory. Can be NULL. Default uses malloc.
-         * @param freeCallback Callback for freeing memory. Can be NULL. Default uses free.
-         */
-        void SetContext(IPLLogFunction logCallback, IPLAllocateFunction allocCallback, IPLFreeFunction freeCallback);
     
-        /// Creates a scene object, to be populated with meshes.
+        /// Creates the scene object, to be populated with meshes.
         /**
          * @param settings The settings to use for simulation.
          * @param numMaterials Number of different materials used. May not be added or removed after Scene is created.
-         * @return Scene object to be used for adding meshes.
-         */
-        IPLhandle* CreateScene(IPLSimulationSettings settings, uint32_t numMaterials);
-
-        /// Destroys a scene object (once reference count reaches zero).
-        /**
-         * @param scene Handle to a scene to be destroyed.
          */
-        void DestroyScene(IPLhandle* scene);
+        void CreateScene(uint32_t numMaterials);
 
-        /// Finalizes a scene object.
+        /// Finalizes the scene object.
         /**
-         * @param scene Handle to the scene to be finalized.
          * @param progressCallback Callback to check the progress of the finalization. Can be NULL.
          */
-        void FinalizeScene(IPLhandle* scene, IPLFinalizeSceneProgressCallback progressCallback); //Can also create Environment (until we start using multiple scenes)
+        void FinalizeScene(IPLFinalizeSceneProgressCallback progressCallback); //Can also create Environment (until we start using multiple scenes)
     
         /// Everything needed to load a previously finalized scene stored in a byte array.
         struct SaveData {
             IPLSimulationSettings settings;
             int sceneSize;
-            char* scene;
+            IPLbyte* scene;
         };
 
-        /// Saves a finalized scene to a struct.
+        /// Saves the finalized scene to a struct.
         /**
-         * @param scene The scene to save
          * @return A struct containing everything needed to reconstruct the scene with LoadFinalizedScene().
          */
-        SaveData SaveFinalizedScene(IPLhandle* scene); //Note to self: Should call iplSaveFinalizedScene with null data pointer first to figure out how big to make the data pointer for a second call
+        SaveData SaveFinalizedScene();
 
         /// Loads a finalized scene from a struct.
         /**
          * @param data Struct containing all data relevant to reconstructing a scene
-         * @return Scene
          */
-        IPLhandle* LoadFinalizedScene(SaveData data);
+        void LoadFinalizedScene(SaveData data);
```

Comment:
Any reason `data` can't be a const reference?

Reply:
Fixed